app/src/components/card/cardImage.tsx

```diff
@@ -0,0 +1,64 @@
+import Image from "next/image";
+import React from "react";
+import { useEffect, useRef, useState } from "react";
+import { isCollectionCardDataType } from "./card";
+
+const CardImage = ({ record }) => {
+  const [imageHeight, setImageHeight] = useState(144);
+  const imageRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    const updateImageHeight = () => {
+      if (imageRef.current) {
+        const image = imageRef.current as HTMLElement;
+        const imageWidth = image.offsetWidth;
+
+        setImageHeight(0.5 * imageWidth);
+      }
+    };
+    updateImageHeight();
+    window.addEventListener("resize", updateImageHeight);
+
+    return () => window.removeEventListener("resize", updateImageHeight);
+  }, []);
```

Comment:
This means this runs for every image. Sometimes, the page feels laggy but I'm not sure if it's because of this. Let's revisit this for a later PR

Reply:
Pulled this out to the lane/grid level (same as the tooltip offset hook) which I had totally forgotten but that had been my plan since I did the tooltip. Actually i think i could go a step further and combine those two hooks since they're using the same thing (the image height) to do some calculations then return a number. What do you guys think, is that confusing to have one hook return two things ? @avertrees @EdwinGuzman

Reply:
It's a bit confusing and ideally a hook would return one value. I'd say it's okay to put them together since they are doing pretty much the same logic, but keep them separate. It'll be two hooks but once you move it up in the DOM tree, it'll be two hooks running instead of the 24+ for each image on the page.

Reply:
I think that's how it is rn unless you mean combining them in one file. But currently 2 hooks on the lane/grid

Reply:
Right, I wasn't too clear but keep them as 2 hooks as it exists now.